__tests__/app.test.js

```diff
@@ -225,3 +225,81 @@ describe("POST /api/articles/:article_id/comments", () => {
     expect(body).toEqual({ msg: "Article not found" });
   });
 });
+
+describe("PATCH /api/articles/:article_id", () => {
+  test(`200: Updates an article's votes by a given increment, 
+    and responds with the updated article object`, async () => {
+    const testIncrements = [
+      { inc_votes: 1 },
+      { inc_votes: 100 },
+      { inc_votes: -101 },
+    ];
+    const testArticleIds = [1, 5, 8];
+    for (const testArticleId of testArticleIds) {
+      let startVote = testData.articleData[testArticleId - 1].votes || 0;
+      for (const testInc of testIncrements) {
+        const {
+          body: { updatedArticle },
+        } = await request(app)
+          .patch(`/api/articles/${testArticleId}`)
+          .send(testInc)
+          .expect(200);
+        expect(updatedArticle).toEqual(
+          expect.objectContaining({
+            article_id: testArticleId,
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            article_img_url: expect.any(String),
+            votes: startVote + testInc.inc_votes,
+          })
+        );
+        const {
+          body: { article },
+        } = await request(app).get(`/api/articles/${testArticleId}`);
+        expect(updatedArticle).toEqual(article);
+        startVote += testInc.inc_votes;
+      }
+    }
+  });
```

Comment:
This is a great approach to testing the PATCH /api/articles/:article_id endpoint. Most students focus on a single request for one article, but you've gone further by dynamically testing multiple articles and increments. This demonstrates a strong understanding of broader test coverage and showcases an innovative method.

Your use of nested loops is an interesting way to avoid duplication while covering multiple scenarios. To refine this, consider resetting breaking out helper functions for readability.

Overall, this is a well-thought-out and thorough test—great work!